crates/services/block_aggregator_api/src/api/protobuf_adapter.rs

```diff
@@ -0,0 +1,185 @@
+                            let response = BlockResponse {
+                                payload: Some(block_response::Payload::Literal(pb)),
+                            };
+                            if tx.send(Ok(response)).await.is_err() {
+                                break;
+                            }
+                        }
+                    });
+
+                    Ok(tonic::Response::new(ReceiverStream::new(rx)))
+                }
+                BlockRangeResponse::Remote(_) => {
+                    tracing::error!("Remote block range not implemented");
+                    todo!()
+                }
+            },
+            Err(e) => Err(tonic::Status::internal(format!(
+                "Failed to receive block range: {}",
+                e
+            ))),
+        }
+    }
+
+    type NewBlockSubscriptionStream = ReceiverStream<Result<BlockResponse, Status>>;
+
+    async fn new_block_subscription(
+        &self,
+        request: tonic::Request<NewBlockSubscriptionRequest>,
+    ) -> Result<tonic::Response<Self::NewBlockSubscriptionStream>, tonic::Status> {
+        const ARB_CHANNEL_SIZE: usize = 100;
+        tracing::warn!("get_block_range: {:?}", request);
+        let (response, mut receiver) = tokio::sync::mpsc::channel(ARB_CHANNEL_SIZE);
+        let query = BlockAggregatorQuery::NewBlockSubscription { response };
+        self.query_sender
+            .send(query)
+            .await
+            .map_err(|e| Status::internal(format!("Failed to send query: {}", e)))?;
+
+        let (task_sender, task_receiver) = tokio::sync::mpsc::channel(ARB_CHANNEL_SIZE);
+        tokio::spawn(async move {
+            while let Some(nb) = receiver.recv().await {
+                let block = Block {
+                    data: nb.block.bytes().to_vec(),
+                };
+                let response = BlockResponse {
+                    payload: Some(block_response::Payload::Literal(block)),
+                };
+                if task_sender.send(Ok(response)).await.is_err() {
+                    break;
+                }
+            }
+        });
+
+        Ok(tonic::Response::new(ReceiverStream::new(task_receiver)))
+    }
+}
+
+pub struct ProtobufAPI {
+    _server_task_handle: tokio::task::JoinHandle<()>,
+    query_receiver: tokio::sync::mpsc::Receiver<BlockAggregatorQuery<BlockRangeResponse>>,
+}
+
+impl ProtobufAPI {
+    pub fn new(url: String) -> Self {
+        let (query_sender, query_receiver) =
+            tokio::sync::mpsc::channel::<BlockAggregatorQuery<BlockRangeResponse>>(100);
+        let server = Server::new(query_sender);
+        let addr = url.parse().unwrap();
+        let _server_task_handle = tokio::spawn(async move {
+            tonic::transport::Server::builder()
+                .add_service(block_aggregator_server::BlockAggregatorServer::new(server))
+                .serve(addr)
+                .await
+                .unwrap();
+        });
```

Comment:
I add handling for this error as well as adding a drop impl for this type so the thread doesn't get orphaned here:
https://github.com/FuelLabs/fuel-core/pull/3101/commits/7f917bab8876f14f9ea2a0cb11d9f23d0089a928